### ISSUE TITLE
Language tagged strings and i18n

### ISSUE THREAD

**Issue by steffansluis** [cayleygraph/cayley#647]
I noticed that language tagged strings are already supported internally in Cayley. It would be nice to be able to use this functionality through the HTTP APIs somehow (mostly Gizmo and GraphQL). I'm not entirely sure yet what would be the best way to include this. I would think it work somewhat similarly to the labeled sub-graphs, so I could imagine a `Lang` directive for Gizmo and a `@lang` and `@language` directive for GraphQL (similar to what I proposed in #614). Maybe it makes more sense though to have it work like the `@opt` directive where it doesn't propagate implicitly to the nested query. In any case, curious to hear what you think about it!

**Comment by dennwc** [comment]
Good point, language strings are implemented for quite some time, but we don't have any queries for them yet. Also, KV/memstore has no indexes for `lang`, while SQL/Mongo can be easily extended to provide support for these queries by enabling an index on this field.

This feature is not implemented mostly because we wanted to postpone this work after reification is done, since `lang` will look like a custom metadata field on string nodes, and all custom metadata queries will work for language as well. But it makes sense to add `@lang` in current implementation, since it's not that hard and internal details about implementation will change under the hood of the query and will not affect GraphQL.

And, more details should be discussed before starting this. For example, what is the default behavior for `name @lang("en")` query in case "en" language string does not exists? Should it assume that there is no such predicate, or fallback to string without language tags? What if only German string is available, but not a generic version?

**Comment by steffansluis** [comment]
[SPARQL](https://www.w3.org/TR/sparql11-query/#matchLangTags) treats language tagged strings and non-language tagged strings as different values because they are not the same RDF literals. This makes  sense, since the [RDF spec](https://www.w3.org/TR/rdf11-concepts/#dfn-language-tag) also states that the datatype differs for language-tagged strings. As such, I would think that when **not** using the `@lang` directive the `name` has to be the regular type of string and with the `@lang` directive the `name` would have to be the language-tagged type of string. Therefore, there is no generic version of a language-tagged string since they are distinct values within their type, so no fallback, just a null result if a particular language doesn't have a value defined. The [specification of the language](https://tools.ietf.org/html/bcp47) is standardized as well by the RDF spec.

**Comment by iddan** [comment]
Hey, I need to get literals language literals in my Gizmo queries but I can't. I tried to look into the code of Gizmo but I was not able to figure out where the values are formatted and why `LangString` is not formatted correctly. @dennwc can you help me out?

**Comment by iddan** [comment]
Found it! My fix is here: https://github.com/cayleygraph/cayley/pull/819

**Comment by iddan** [comment]
@dennwc  as far as I understand we currently load language strings correctly from files and when inserting quads through the UI.

**Comment by dennwc** [comment]
@iddan Correct. The only issues is a query-side support. Right now it's possible to query for exact match, but not with a wildcard in the language, for example.

**Comment by iddan** [comment]
Got it.

**Comment by iddan** [comment]
https://github.com/cayleygraph/cayley/pull/834 solved querying for lang strings with regex filter
